### DISCUSSION TITLE
Autocomplete parameters

### DISCUSSION THREAD

**Discussion by cshjsc** [mfussenegger/nvim-jdtls#440] [General]
Hi, I'm not sure if this is a problem with nvim-jdtls or not but when I select a method with autocomplete (nvim-cmp) I don't get the parameters filled automatically like in other languages.
Instead of for example myMethod(a,b,c). I just get mymethod . I don't know if its a problem with nvim-cmp or with the lsp that doesn't provide the information to cmp.

**Comment by mfussenegger** [comment]
You have to setup snippet support for your completion plugin.


https://user-images.githubusercontent.com/38700/224793477-2924d0dd-e534-4ce9-9da1-e7d6f653837c.mp4

**Comment by soeguet** [comment]
I have the same problem and I am pretty confused. using jdtls without this plugin gives me the wanted autocompletion (with a '~' in the completion pop up/drop down). 

at what point do I need to include nvim-jdtls into cmp? Is there any .lua example for that?

**Comment by n-xiao** [comment]
@mfussenegger I am having the same issue... Could you please clarify again? The video you sent is marked as corrupted.

**Comment by n-xiao** [comment]
### Got it working
How to enable autocomplete for parameters (arguments are autofilled with placeholders, you are able to use TAB to navigate / jump to next argument && ENTER to change placeholder value)

You have to add **snippentSupport** in your nvim/ftplugin/java.lua

```
local config = {
  ...
  capabilities = {
    workspace = {
      configuration = true,
    },
    textDocument = {
      completion = {
        completionItem = {
          snippentSupport = true,
        },
      },
    },
  },
}
config.capabilities = require("cmp_nvim_lsp").default_capabilities(vim.lsp.protocol.make_client_capabilities())
require("jdtls").start_or_attach(config)
```

**Reply by PeytonSch** [reply]
Working great thanks!

**Reply by rogedelgado** [reply]
Hello!

I faced the same problem, and implementing the suggested config made it work, but I just needed to add the following line:

`config.capabilities = require("cmp_nvim_lsp").default_capabilities(vim.lsp.protocol.make_client_capabilities())`

Is the "capability" section in the config really needed?

> ```
> local config = {
>   ...
>   capabilities = {
>     workspace = {
>       configuration = true,
>     },
>     textDocument = {
>       completion = {
>         completionItem = {
>           snippentSupport = true,
>         },
>       },
>     },
>   },
> }
> config.capabilities = require("cmp_nvim_lsp").default_capabilities(vim.lsp.protocol.make_client_capabilities())
> require("jdtls").start_or_attach(config)
> ```

**Reply by n-xiao** [reply]
Hello, I know it's been quite a while but I think I've figured it all out. When I first posted my solution, I had no idea what I was doing and it was a miracle that it even worked. Just putting this stuff out here so I don't spread wrong information.

1.  the capabilities part is required, but the old solution had unnecessary code.
2. `textDocument = {...}` is totally unnecessary.

**What you do need:**
1. A snippet plugin, like vsnip that has properly configured keybinds.
2. Assuming you use nvim-lspconfig with nvim-cmp (a completion plugin), you need to add the capabilities to the config:

```
local capabilities = require('cmp_nvim_lsp').default_capabilities()
local config = {
    capabilities = capabilities,
    ...
}
...
```

hope this clears things up
